CaptureSDK/GiniCaptureSDK/Documentation/source/Migration to version 3.x.x.md

```diff
@@ -1,8 +1,178 @@
+
+# Overview of New UI Customization Options
+
+To simplify UI customization we introduced global customization options. There is no need to customize each screen separately anymore.
+
+## Colors
+
+We are providing a global color palette `GiniColors.xcassets` which you are free to override. The custom colors will be then applied on all screens.
+You can find the names of the colors in [GiniColors.xcassets](https://github.com/gini/gini-mobile-ios/tree/new-ui/CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Resources/GiniColors.xcassets).
+
+ You can view our color palette here:
+
+<iframe style="border: 1px solid rgba(0, 0, 0, 0.1);" width="600" height="450" src="https://www.figma.com/embed?embed_host=share&url=https%3A%2F%2Fwww.figma.com%2Ffile%2FH4WFJ4xaw4YNU4VaJYWiQq%2FiOS-Gini-Capture-SDK-2.0.0-UI-Customisation%3Fnode-id%3D14%253A355%26t%3DwpenBBM8QsagJzOg-1" allowfullscreen></iframe>
+
+## Typography
+
+We provide a global typography based on text appearance styles from `UIFont.TextStyle`. 
+
+<iframe style="border: 1px solid rgba(0, 0, 0, 0.1);" width="600" height="450" src="https://www.figma.com/embed?embed_host=share&url=https%3A%2F%2Fwww.figma.com%2Ffile%2FH4WFJ4xaw4YNU4VaJYWiQq%2FiOS-Gini-Capture-SDK-2.0.0-UI-Customisation%3Fnode-id%3D123%253A2326%26t%3DwpenBBM8QsagJzOg-1" allowfullscreen></iframe>
+
+To override them in your application please use `GiniConfiguration.updateFont(_ font: UIFont, for textStyle: UIFont.TextStyle)`. For example:
+
+```swift
+
+    // If you need to scale your font please use our method `scaledFont()`. Please, find the example below.
+    let configuration = GiniBankConfiguration.shared
+    let customFontToBeScaled = UIFont.scaledFont(UIFont(name: "Avenir", size: 20) ?? UIFont.systemFont(ofSize: 7, weight: .regular), textStyle: .caption1)
+    configuration.updateFont(customFontToBeScaled, for: .caption1)
+
+    // If you would like to pass us already scaled font.
+    let customScaledFont = UIFontMetrics(forTextStyle: .caption2).scaledFont(for: UIFont.systemFont(ofSize: 28))
+    configuration.updateFont(customScaledFont, for: .caption2)
+
+```
+## Images
+
+Images customization is done via overriding of [GiniImages.xcassets](https://github.com/gini/gini-mobile-ios/tree/new-ui/CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Resources/GiniImages.xcassets) resources.
+
+## Text
+
+ Text customization is done via overriding of string resources.
+
+ You can find all the string resources in [Localizable.strings](https://github.com/gini/gini-mobile-ios/blob/new-ui/CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Resources/de.lproj/Localizable.strings).
+
+# UI Elements
+
+Certain elements of the UI can now be fully customized via UI injection. It utilizes view adapter interfaces which you
```

Comment:
[Suggestion] As we have some of them implemented in the example app, shouldn't we mention it that they can look into that?

Reply:
Yep, very nice. I'll link it. 😊 Thank you.